store.go

```diff
@@ -1470,7 +1470,7 @@ func (s *store) putLayer(rlstore rwLayerStore, rlstores []roLayerStore, id, pare
 	if lOptions != nil {
 		options = *lOptions
 		options.BigData = copyLayerBigDataOptionSlice(lOptions.BigData)
-		options.Flags = maps.Clone(lOptions.Flags)
+		options.Flags = copyStringInterfaceMap(lOptions.Flags)
```

Comment:
I think this is unnecessary here; `nil`ness of `Flags` in itself carries no meaning, and the consumer in `layerStore.Create` is already using `copyStringInterfaceMap` to guarantee a non-`nil` value it needs.